NewBlackAuthenticator/Controllers/TwoFARegistrationsController.cs

```diff
@@ -0,0 +1,201 @@
+        {
+            if (id != twoFARegistration.Id)
+            {
+                return BadRequest();
+            }
+
+            _context.Entry(twoFARegistration).State = EntityState.Modified;
+
+            try
+            {
+                await _context.SaveChangesAsync();
+            }
+            catch (DbUpdateConcurrencyException)
+            {
+                if (!TwoFARegistrationExists(id))
+                {
+                    return NotFound();
+                }
+                else
+                {
+                    throw;
+                }
+            }
+
+            return NoContent();
+        }
+
+        // POST: api/TwoFARegistrations
+        // To protect from overposting attacks, enable the specific properties you want to bind to, for
+        // more details, see https://go.microsoft.com/fwlink/?linkid=2123754.
+        [HttpPost]
+        public async Task<ActionResult<TwoFARegistration>> PostTwoFARegistration(TwoFARegistration twoFARegistration)
+        {
+            string token = RandomTokenGenerator(32);
+            twoFARegistration.Token = token;
+
+            _context.TwoFARegistrations.Add(twoFARegistration);
+            try
+            {
+                await _context.SaveChangesAsync();
+            }
+            catch (DbUpdateException)
+            {
+                if (TwoFARegistrationExists(twoFARegistration.Id))
+                {
+                    return Conflict();
+                }
+                else
+                {
+                    throw;
+                }
+            }
+
+            return CreatedAtAction("GetTwoFARegistration", new { id = twoFARegistration.Id }, twoFARegistration);
+        }
+
+        // DELETE: api/TwoFARegistrations/5
+        [HttpDelete("{id}")]
+        public async Task<ActionResult<TwoFARegistration>> DeleteTwoFARegistration(string id)
+        {
+            var twoFARegistration = await _context.TwoFARegistrations.FindAsync(id);
+            if (twoFARegistration == null)
+            {
+                return NotFound();
+            }
+
+            _context.TwoFARegistrations.Remove(twoFARegistration);
+            await _context.SaveChangesAsync();
+
+            return twoFARegistration;
+        }
+
+        private bool TwoFARegistrationExists(string id)
+        {
+            return _context.TwoFARegistrations.Any(e => e.Id == id);
+        }
+
+        //api/TwoFARegistrations
+        [HttpPost("UpdateTwoFARegistrationStatus")]
+        public ActionResult<TwoFARegistration> UpdateTwoFARegistrationStatus(UpdateTwoFARegistrationStatusRequest updateStatusReguest)
+        {
+            var token = updateStatusReguest.Token;
+            var userAnswer = updateStatusReguest.UserAnswer;
+
+                var twoFAReg = _context.TwoFARegistrations
+                .Where(r => r.Token == token)
+                .FirstOrDefault();
+
+            // how to use enum here??? // 
```

Comment:
I don't know what you're trying to achieve! Poke me tomorrow and I'll see if I can write up some Linqpad code to get you on track.